### PR TITLE
chore(feedback): Remove links to deprecated submit-user-feedback endpoint

### PR DESCRIPTION
**Pull request by aliu39**
Ref https://github.com/getsentry/sentry-docs/issues/10643
Relates to https://github.com/getsentry/sentry/pull/75396

- kept links in the develop docs and ruby on rails, since it doesn't have an SDK API documented
- I didn't find any links to the list-a-projects-user-feedback (GET) endpoint.

### CODE REVIEW

File: develop-docs/application/feedback-architecture.mdx [hunk 1]
```diff
@@ -4,10 +4,12 @@ title: User Feedback Architecture
 
 **The goal of this doc is to give engineers an in-depth understanding of User Feedback's backend.**
 It will:
+
```
Comment (aliu39) [inline develop-docs/application/feedback-architecture.mdx:7]: Extra lines were added by my formatter but I'm not worried about them

**Comment by michellewzhang** [comment]
note for myself: we'll also need to remove references to these links in feedback onboarding

File: docs/platforms/javascript/common/user-feedback/index.mdx [hunk 2]
```diff
@@ -38,24 +37,31 @@ By default, this will insert the widget into the bottom right corner of your web
 On SDK version 8.0.0 and above, users can send screenshots with their feedback. If you're self-hosting, you also need release 24.4.2 and above. You can configure this using the `enableScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.
 
 <Alert level="">
-  Screenshots use your [attachments quota](/pricing/quotas/manage-attachments-quota). All plans come with 1GB of attachments, which is approxiamately 2500 screenshots.
+  Screenshots use your [attachments
```
Comment (michellewzhang) [inline docs/platforms/javascript/common/user-feedback/index.mdx:40]: was this one also automatically added?

File: docs/platforms/javascript/common/user-feedback/index.mdx [hunk 3]
```diff
@@ -38,24 +37,31 @@ By default, this will insert the widget into the bottom right corner of your web
...
 
 </PlatformSection>
 
 ## User Feedback API
 
 The user feedback API allows you to collect user feedback while utilizing your own UI. You can use the same programming language you have in your app to send user feedback. In this case, the SDK creates the HTTP request so you don't have to deal with posting data via HTTP.
 
-Sentry pairs the feedback with the original event, giving you additional insight into issues. Sentry needs the `eventId` to be able to associate the user feedback to the corresponding event. For example, to get the `eventId`, you can use <PlatformLink to="/configuration/options/#before-send"><PlatformIdentifier name="before-send" /></PlatformLink> or the return value of the method capturing an event.
+### Stand-alone feedback
```
Comment (michellewzhang) [inline docs/platforms/javascript/common/user-feedback/index.mdx:55]: no content under this heading

File: docs/platforms/javascript/common/user-feedback/index.mdx (hunk 2 above)
Comment (aliu39) [reply docs/platforms/javascript/common/user-feedback/index.mdx:40]: Great catch! I accidentally committed this file, it has nothing to do with this PR/issue 😅

**Review by michellewzhang — APPROVED** [review]

**Review by c298lee — APPROVED** [review]

**Pull request merged**